proofs/flatten.go

```diff
@@ -29,24 +29,24 @@ type messageFlattener struct {
 	valueEncoder      ValueEncoder
 	compactProperties bool
 }
-type GenerateSalt func() []byte
+type GetSalt func(compactHextString string, saltsMap InnerCompactsSaltsMap) []byte
 
-func (f *messageFlattener) handleValue(prop Property, value reflect.Value,  generateSalt GenerateSalt, saltsLengthSuffix string, outerFieldDescriptor *go_descriptor.FieldDescriptorProto) (err error) {
+func (f *messageFlattener) handleValue(prop Property, value reflect.Value,  getSalt GetSalt, saltsMap InnerCompactsSaltsMap, saltsLengthSuffix string, outerFieldDescriptor *go_descriptor.FieldDescriptorProto) (err error) {
```

Comment:
Why does a `GetSalt` function need to be provided at all if the structure fo saltsMap is fixed anyway? I think we can either:
* make the implementation of GetSalt fixed and not allow passing it in. This is much simpler as it only requires a saltsMap to be initialized and passed in.
* remove the saltsMap from the method here and require the GetSalt implementation to have a pointer to it inline (this would allow more flexibility if you would want to store the salts in a different way)

Personally I prefer the first approach as its simpler and requires less work for the implementor.

Reply:
main reason of introducing GetSalt is for unit testing (exist unit test use a fixed salt), If GetSalt is removed, just keep saltsMap, then most of the current unit test need fix, you need generate fixed salt for every fields (means configure compacts slat mapping) in most unit test. It's a huge work.

Reply:
Also, future unit test coding need much work.

Reply:
But if you mean `handleValue `should just contain one augument which combile `GetSalt `and `SaltsMap`, then it is fine

Reply:
Yes